### PR TITLE
Added ConfigException while using not exists list_id

### PR DESCRIPTION
**Pull request by thangnc**
- Catch `HttpResponseException` for 404 status code
- Throws `ConfigException` while using not exists `list_id`

### CODE REVIEW

File: src/main/java/org/embulk/output/mailchimp/MailChimpHttpClient.java [hunk 1]
```diff
@@ -64,36 +65,46 @@ public JsonNode sendRequest(final String endpoint, final HttpMethod method, fina
...
-                        return status == 429 || status / 100 != 4;
-                    }
-                });
+            return responseBody != null && !responseBody.isEmpty() ? parseJson(responseBody) : MissingNode.getInstance();
+        }
+        catch (HttpResponseException hre) {
+            LOG.error("Exception occurred while sending request: {}", hre.getMessage());
+            if (hre.getResponse().getStatus() == 404) {
+                throw new ConfigException("The `list id` could not be found");
+            }
```
Comment (sakama) [inline src/main/java/org/embulk/output/mailchimp/MailChimpHttpClient.java]: I think you can write as following.
```java
 @Override
public boolean isResponseStatusToRetry(Response response)
{
    int status = response.getStatus();
    if (status == 404) {
        LOG.error("Exception occurred while sending request: {}", hre.getMessage());
        throw new ConfigException("The `list id` could not be found");
    }
    return status == 429 || status / 100 != 4;
}
```

Comment (thangnc) [reply src/main/java/org/embulk/output/mailchimp/MailChimpHttpClient.java]: Oh, this way is better than mine.

**Comment by thangnc** [comment]
@sakama PTAL

**Review by sakama — APPROVED** [review]
LGTM

**Pull request merged**